File: gopaw/examples/users/read.c
/*
 * Signature must be on one line starting with "function" in C comment. 
 * Argument left parenthesis must stick the function name.
 */

#include <stdio.h>

/* RUN,ACT,EACT are VECTORs
   [] assumes that the vector already exists. 
   [n] check that a vector TYP exists :
         if the existing VECTOR is not of dim n, we have an error.
         if the existing VECTOR is of dime n, it is ok and it is used.
       if a vector TYP does not exist, it is created with dimension n.
*/

/*function*/ double read(int K,double RUN[],double ACT[],double EACT[]) {      
      int I,J,IRU;
      float AC,EAC;
      FILE* file;

/*
      printf("debug : read : K %d\n",K);
*/

      file = fopen("activite_radon_gas_trace_info","rb");
      if(file==NULL) {
        printf("Can't open activite_radon_gas_trace_info\n");
        return 0;
      }
 
      I = 0;

    /*J = 0;*/
      J = -1;
      while(!feof(file)) {
        if(fscanf(file,"%d %g %g\n",&IRU,&AC,&EAC)!=3) {
          printf("Erreur de lecture de activite_radon_gas_trace_info\n");
          fclose(file);
          return 0;
        }
/*
        printf(" Lecture : IRU %d AC %g EAC %g\n",IRU,AC,EAC);
*/
        J = J + 1;
        if( (IRU>=(3580+(K-1)*200)) && (IRU<(3580+K*200)) ) {
          RUN[J] = (double)IRU;
          ACT[J] = AC;
          EACT[J] = EAC;
        }
/*
        printf(" J %d RUN(J) %g ACT(J) %g EACT(J) %g\n",
          J,RUN[J],ACT[J],EACT[J]);
*/
      }

      fclose(file);

      return 0;
}

File: gopaw/examples/users/read.py

# RUN,ACT,EACT are VECTORs
#   [] assumes that the vector already exists. 
#   [n] check that a vector TYP exists :
#       if the existing VECTOR is not of dim n, we have an error.
#       if the existing VECTOR is of dim n, it is ok and it is used.
#       if a vector TYP does not exist, it is created with dimension n.

# the Python signature contains at end, in a Python comment, the C signature to help declare VECTORs.

def read(K,RUN,ACT,EACT):  #/*function*/ double read(int K,double RUN[],double ACT[],double EACT[])
 #print 'debug : read.py : K ',K
  
  file = open("data_activite_radon_gas_trace.dat","rb")
 
  I = 0

 #J = 0
  J = -1
  for line in file:
    words = line.split()
    #print words
    IRU = int(words[0])    #   int() is important.
    AC  = float(words[1])  # float() is important.
    EAC = float(words[2])  # float() is important.
    
   #print ' Lecture : IRU ',IRU,' AC ',AC,' EAC ',EAC
   
    J = J + 1
    if (IRU>=(3580+(K-1)*200)) and (IRU<(3580+K*200)) :
      RUN[J]  = IRU
      ACT[J]  = AC
      EACT[J] = EAC
    
   #print ' J ',J,' RUN(J) ',RUN[J],' ACT(J) ',ACT[J],' EACT(J) ',EACT[J]

  file.close()
